File: app/src/main/java/com/example/protect/ui/screens/OrdersScreen.kt
package com.example.protect.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.protect.model.OrderItem
import com.example.protect.model.OrderStatus
import com.example.protect.theme.*
import com.example.protect.ui.components.ProtectTopBar
import com.example.protect.ui.components.StatusBadge
import com.example.protect.viewmodel.AppViewModel
import java.text.NumberFormat
import java.util.*

@Composable
fun OrdersScreen(
    viewModel: AppViewModel,
    onNavigateToLiveTracker: () -> Unit,
    onNavigateToWarranty: () -> Unit,
    onNavigateToBooking: () -> Unit
) {
    val orders by viewModel.orders.collectAsState()
    var selectedFilter by remember { mutableStateOf("Semua") }

    val filters = listOf("Semua", "Sedang Berjalan", "Selesai")

    val filteredOrders = orders.filter { order ->
        when (selectedFilter) {
            "Sedang Berjalan" -> order.status == OrderStatus.EN_ROUTE || order.status == OrderStatus.IN_PROGRESS || order.status == OrderStatus.PENDING
            "Selesai" -> order.status == OrderStatus.COMPLETED
            else -> true
        }
    }

    val currencyFormat = remember {
        NumberFormat.getCurrencyInstance(Locale("id", "ID")).apply {
            maximumFractionDigits = 0
        }
    }

    Scaffold(
        topBar = {
            ProtectTopBar(
                title = "Riwayat Pesanan Layanan"
            )
        }
    ) { innerPadding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .background(BackgroundLight)
                .padding(innerPadding),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            // Filter Pills
            item {
                LazyRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    items(filters) { f ->
                        val isSelected = f == selectedFilter
                        Surface(
                            shape = RoundedCornerShape(16.dp),
                            color = if (isSelected) Emerald600 else SurfaceLight,
                            modifier = Modifier
                                .border(1.dp, if (isSelected) Emerald600 else BorderColor, RoundedCornerShape(16.dp))
                                .clickable { selectedFilter = f }
                        ) {
                            Text(
                                text = f,
                                fontSize = 12.sp,
                                fontWeight = FontWeight.SemiBold,
                                color = if (isSelected) SurfaceLight else TextSecondary,
                                modifier = Modifier.padding(horizontal = 14.dp, vertical = 8.dp)
                            )
                        }
                    }
                }
            }

            if (filteredOrders.isEmpty()) {
                item {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(top = 40.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Icon(
                            imageVector = Icons.Default.AssignmentLate,
                            contentDescription = null,
                            tint = TextMuted,
                            modifier = Modifier.size(64.dp)
                        )
                        Spacer(modifier = Modifier.height(12.dp))
                        Text(
                            text = "Tidak ada pesanan pada kategori ini.",
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Bold,
                            color = TextSecondary
                        )
                    }
                }
            } else {
                items(filteredOrders) { order ->
                    Card(
                        shape = RoundedCornerShape(16.dp),
                        colors = CardDefaults.cardColors(containerColor = SurfaceLight),
                        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
                        modifier = Modifier
                            .fillMaxWidth()
                            .border(1.dp, BorderColor, RoundedCornerShape(16.dp))
                    ) {
                        Column(modifier = Modifier.padding(16.dp)) {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Text(
                                    text = order.orderNumber,
                                    fontSize = 12.sp,
                                    fontWeight = FontWeight.Bold,
                                    color = Emerald600
                                )
                                StatusBadge(
                                    text = when (order.status) {
                                        OrderStatus.EN_ROUTE -> "EN ROUTE"
                                        OrderStatus.IN_PROGRESS -> "DALAM PROSES"
                                        OrderStatus.COMPLETED -> "SELESAI & BERGARANSI"
                                        OrderStatus.PENDING -> "MENUNGGU"
                                        OrderStatus.CANCELLED -> "DIBATALKAN"
                                    },
                                    backgroundColor = when (order.status) {
                                        OrderStatus.EN_ROUTE -> Emerald50
                                        OrderStatus.COMPLETED -> Emerald100
                                        else -> SurfaceVariantLight
                                    },
                                    textColor = when (order.status) {
                                        OrderStatus.EN_ROUTE -> Emerald700
                                        OrderStatus.COMPLETED -> Emerald900
                                        else -> TextSecondary
                                    }
                                )
                            }

                            Spacer(modifier = Modifier.height(10.dp))

                            Text(
                                text = order.serviceTitle,
                                fontSize = 15.sp,
                                fontWeight = FontWeight.Bold,
                                color = DarkSlate900
                            )

                            Spacer(modifier = Modifier.height(6.dp))

                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(
                                    imageVector = Icons.Default.CalendarToday,
                                    contentDescription = null,
                                    tint = TextSecondary,
                                    modifier = Modifier.size(13.dp)
                                )
                                Spacer(modifier = Modifier.width(6.dp))
                                Text(
                                    text = "${order.date} • ${order.time}",
                                    fontSize = 11.sp,
                                    color = TextSecondary
                                )
                            }

                            Spacer(modifier = Modifier.height(4.dp))

                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(
                                    imageVector = Icons.Default.LocationOn,
                                    contentDescription = null,
                                    tint = TextSecondary,
                                    modifier = Modifier.size(13.dp)
                                )
                                Spacer(modifier = Modifier.width(6.dp))
                                Text(
                                    text = order.address,
                                    fontSize = 11.sp,
                                    color = TextSecondary,
                                    maxLines = 1
                                )
                            }

                            HorizontalDivider(modifier = Modifier.padding(vertical = 12.dp))

                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Column {
                                    Text(
                                        text = "Total Biaya",
                                        fontSize = 11.sp,
                                        color = TextSecondary
                                    )
                                    Text(
                                        text = currencyFormat.format(order.price),
                                        fontSize = 14.sp,
                                        fontWeight = FontWeight.Bold,
                                        color = DarkSlate900
                                    )
                                }

                                if (order.status == OrderStatus.EN_ROUTE) {
                                    Button(
                                        onClick = onNavigateToLiveTracker,
                                        colors = ButtonDefaults.buttonColors(containerColor = Emerald600),
                                        shape = RoundedCornerShape(10.dp)
                                    ) {
                                        Icon(Icons.Default.Navigation, contentDescription = null, modifier = Modifier.size(14.dp))
                                        Spacer(modifier = Modifier.width(6.dp))
                                        Text("Live GPS", fontSize = 12.sp, fontWeight = FontWeight.Bold)
                                    }
                                } else if (order.status == OrderStatus.COMPLETED) {
                                    Button(
                                        onClick = onNavigateToWarranty,
                                        colors = ButtonDefaults.buttonColors(containerColor = Emerald50, contentColor = Emerald700),
                                        shape = RoundedCornerShape(10.dp)
                                    ) {
                                        Icon(Icons.Default.Verified, contentDescription = null, modifier = Modifier.size(14.dp))
                                        Spacer(modifier = Modifier.width(6.dp))
                                        Text("E-Garansi", fontSize = 12.sp, fontWeight = FontWeight.Bold)
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

File: lib/screens/orders_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../data/sample_data.dart';
import '../models/data_models.dart';
import '../theme/app_theme.dart';
import '../widgets/common_widgets.dart';
import 'live_tracker_screen.dart';
import 'digital_warranty_screen.dart';

class OrdersScreen extends StatefulWidget {
  const OrdersScreen({super.key});

  @override
  State<OrdersScreen> createState() => _OrdersScreenState();
}

class _OrdersScreenState extends State<OrdersScreen> {
  int _selectedFilterIndex = 0;
  final List<String> _filters = ['Semua', 'Sedang Berjalan', 'Selesai', 'Garansi Aktif'];
  final NumberFormat _currencyFormat = NumberFormat.currency(locale: 'id_ID', symbol: 'Rp ', decimalDigits: 0);

  @override
  Widget build(BuildContext context) {
    final orders = SampleData.getInitialOrders();
    final filtered = orders.where((ord) {
      if (_selectedFilterIndex == 1) return ord.status == OrderStatus.enRoute || ord.status == OrderStatus.inProgress;
      if (_selectedFilterIndex == 2) return ord.status == OrderStatus.completed;
      if (_selectedFilterIndex == 3) return ord.warrantyUntil.isNotEmpty;
      return true;
    }).toList();

    return Scaffold(
      backgroundColor: AppColors.backgroundLight,
      appBar: AppBar(
        title: const Text('Daftar Pesanan & Riwayat Treatment'),
      ),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          // Filter Tabs
          SingleChildScrollView(
            scrollDirection: Axis.horizontal,
            child: Row(
              children: List.generate(_filters.length, (i) {
                final isSelected = _selectedFilterIndex == i;
                return Padding(
                  padding: const EdgeInsets.only(right: 8),
                  child: ChoiceChip(
                    label: Text(_filters[i]),
                    selected: isSelected,
                    onSelected: (val) => setState(() => _selectedFilterIndex = i),
                    selectedColor: AppColors.emerald600,
                    labelStyle: TextStyle(
                      color: isSelected ? Colors.white : AppColors.textSecondary,
                      fontWeight: FontWeight.bold,
                      fontSize: 12,
                    ),
                    backgroundColor: AppColors.surfaceLight,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(20),
                      side: BorderSide(
                        color: isSelected ? AppColors.emerald600 : AppColors.borderColor,
                      ),
                    ),
                  ),
                );
              }),
            ),
          ),
          const SizedBox(height: 16),

          ...filtered.map((ord) {
            final isEnRoute = ord.status == OrderStatus.enRoute;
            return Container(
              margin: const EdgeInsets.only(bottom: 14),
              decoration: BoxDecoration(
                color: AppColors.surfaceLight,
                borderRadius: BorderRadius.circular(16),
                border: Border.all(color: AppColors.borderColor),
              ),
              padding: const EdgeInsets.all(16),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Text(
                        ord.orderNumber,
                        style: const TextStyle(
                          fontSize: 12,
                          fontFamily: 'monospace',
                          fontWeight: FontWeight.bold,
                          color: AppColors.emerald600,
                        ),
                      ),
                      StatusBadge(
                        text: isEnRoute ? 'TEKNISI MENUJU LOKASI' : 'SELESAI (TERLINDUNGI)',
                        backgroundColor: isEnRoute ? AppColors.emerald50 : AppColors.emerald100.withOpacity(0.5),
                        textColor: isEnRoute ? AppColors.emerald700 : AppColors.emerald900,
                      ),
                    ],
                  ),
                  const SizedBox(height: 10),
                  Text(
                    ord.serviceTitle,
                    style: const TextStyle(fontSize: 15, fontWeight: FontWeight.bold, color: AppColors.darkSlate900),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    'Jadwal: ${ord.date} • ${ord.time}',
                    style: const TextStyle(fontSize: 12, color: AppColors.textSecondary),
                  ),
                  const SizedBox(height: 2),
                  Text(
                    'Alamat: ${ord.address}',
                    style: const TextStyle(fontSize: 12, color: AppColors.textSecondary),
                  ),
                  const Divider(height: 24),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          const Text('Total Pembayaran', style: TextStyle(fontSize: 10, color: AppColors.textSecondary)),
                          Text(
                            _currencyFormat.format(ord.price),
                            style: const TextStyle(fontSize: 14, fontWeight: FontWeight.bold, color: AppColors.darkSlate900),
                          ),
                        ],
                      ),
                      if (isEnRoute)
                        ElevatedButton.icon(
                          onPressed: () {
                            Navigator.of(context).push(
                              MaterialPageRoute(builder: (_) => const LiveTrackerScreen()),
                            );
                          },
                          icon: const Icon(Icons.navigation, size: 14, color: Colors.white),
                          label: const Text('Live GPS Tracking', style: TextStyle(fontSize: 11, color: Colors.white)),
                          style: ElevatedButton.styleFrom(
                            backgroundColor: AppColors.emerald600,
                            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                            padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
                          ),
                        )
                      else
                        OutlinedButton.icon(
                          onPressed: () {
                            Navigator.of(context).push(
                              MaterialPageRoute(builder: (_) => const DigitalWarrantyScreen()),
                            );
                          },
                          icon: const Icon(Icons.verified, size: 14, color: AppColors.emerald600),
                          label: const Text('E-Garansi 12 Bln', style: TextStyle(fontSize: 11, color: AppColors.emerald600)),
                          style: OutlinedButton.styleFrom(
                            side: const BorderSide(color: AppColors.emerald600),
                            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                            padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
                          ),
                        ),
                    ],
                  ),
                ],
              ),
            );
          }),
        ],
      ),
    );
  }
}
